File: VisitCard.js
const MultipleItem = require("./MultipleItem");
const prompt = require("prompt-sync")();

class VisitCard extends MultipleItem {
  constructor(companyName, post, version, quantity, name) {
    super(name, unity);
    this.companyName = companyName;
    this.post = post;
    this.version = version;
    this.quantity = quantity;
  }

  static createVisitCard() {
    const companyName = prompt(
      "Entrez le nom de l'entreprise sur la carte de visite : ",
    );
    const post = prompt("Entrez le poste sur la carte de visite : ");
    const version = prompt("Entrez la version de la carte de visite : ");
    const quantity = parseInt(
      prompt("Entrez la quantité de cartes de visite : "),
      10,
    );

    // Vous pouvez ajouter une logique de validation pour la quantité si nécessaire

    return new VisitCard(companyName, post, version, quantity, "name");
  }

  displayInfo() {
    console.log("Informations de la carte de visite :");
    console.log(`Nom de l'entreprise : ${this.companyName}`);
    console.log(`Poste : ${this.post}`);
    console.log(`Version : ${this.version}`);
    console.log(`Quantité : ${this.quantity}`);
  }
}

module.exports = VisitCard;

File: VisitCard.py
from MultipleItem import MultipleItem
from UtilsMethod import UtilsMethod


class VisitCard(MultipleItem):
    def __init__(self, company_name, post, version, quantity, name):
        super().__init__(name, "unités", quantity)
        self.company_name = company_name
        self.post = post
        self.version = version

    @classmethod
    def from_user_input():
        print("Enter company name: ")
        company_name = input()
        print("Enter post: ")
        post = input()
        print("Enter version: ")
        version = input()
        print("Enter quantity: ")
        quantity = UtilsMethod.take_int_value_from_string(input())

        return VisitCard(company_name, post, version, quantity, "carte de visite")

    def __str__(self):
        return (
            f"VisitCard{{companyName='{self.company_name}', post='{self.post}', "
            f"version='{self.version}', quantity={self.get_quantity()}}}"
        )
